lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -11,15 +11,21 @@ def sync
         deployment_data = JSON.parse(raw_json)
         @desired_replicas = deployment_data["spec"]["replicas"].to_i
         @latest_rs = find_latest_rs(deployment_data)
+
         @rollout_data = { "replicas" => 0 }.merge(deployment_data["status"]
           .slice("replicas", "updatedReplicas", "availableReplicas", "unavailableReplicas"))
         @status = @rollout_data.map { |state_replicas, num| "#{num} #{state_replicas.chop.pluralize(num)}" }.join(", ")
+
         conditions = deployment_data.fetch("status", {}).fetch("conditions", [])
-        @progress = conditions.find { |condition| condition['type'] == 'Progressing' }
+        @progress_condition = conditions.find { |condition| condition['type'] == 'Progressing' }
+        @progress_deadline = deployment_data['spec']['progressDeadlineSeconds']
       else # reset
         @latest_rs = nil
         @rollout_data = { "replicas" => 0 }
         @status = nil
+        @progress_condition = nil
+        @progress_deadline = @definition['spec']['progressDeadlineSeconds']
```

Comment:
Will this always be present in the spec (it's filled by server side defaults)?

Reply:
If it is filled by server-side defaults, it will be in the live copy (i.e. `deployment_data` above) but not this one (`@definition`, which is just the template the user gave us). I figured that if we couldn't access the server version for some reason, getting what the template had is slightly better than defaulting to `nil`.